File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js

function solution(babbling) {
    
    
    var answer = 0; 
    
    const BABB  = ["aya", "ye", "woo", "ma"]  
    
    for(let i = 0 ; i< babbling.length; i++){
        for(let j = 0 ; j< BABB.length ;j ++){
            
            if(babbling[i].includes(BABB[j].repeat(2))){
                break
            }
            babbling[i] =  babbling[i].replaceAll(BABB[j],"1")
        }
    }
    
    
    
    
    
    
    return babbling.map(v=>v.replaceAll("1","")).filter(v=>v==="").length;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        String[] bab = { "aya", "ye", "woo", "ma" }  ; 
        
        for(int i = 0 ; i<babbling.length; i++){
            for(int j= 0 ; j<bab.length; j++ ){
                if(babbling[i].contains(bab[j].repeat(2))){
                    break;
                }
                babbling[i] = babbling[i].replaceAll(bab[j],"1") ;
            }
        }
        for(int i = 0 ; i< babbling.length ; i++){
            babbling[i] = babbling[i].replaceAll("1","") ; 
        }
         for(int i = 0 ; i< babbling.length ; i++){
           if( babbling[i].equals("") ){
               answer++ ; 
           }
        }
        return answer;
    }
}